kafka-ui-react-app/src/components/Topics/Topic/ConsumerGroups/TopicConsumerGroups.tsx

```diff
@@ -5,14 +5,27 @@ import useAppParams from 'lib/hooks/useAppParams';
 import { useTopicConsumerGroups } from 'lib/hooks/api/topics';
 import { ColumnDef } from '@tanstack/react-table';
 import Table, { LinkCell, TagCell } from 'components/common/NewTable';
+import Search from 'components/common/Search/Search';
+
+import * as S from './TopicConsumerGroups.styled';
 
 const TopicConsumerGroups: React.FC = () => {
+  const [keyword, setKeyword] = React.useState('');
   const { clusterName, topicName } = useAppParams<RouteParamsClusterTopic>();
 
-  const { data: consumerGroups = [] } = useTopicConsumerGroups({
+  const { data = [] } = useTopicConsumerGroups({
     clusterName,
     topicName,
   });
+
+  const getData = () => {
+    return data.filter(
+      (item) => item.groupId.toLocaleLowerCase().indexOf(keyword) > -1
+    );
+  };
+
+  const consumerGroups = React.useMemo(() => getData(), [data, keyword]);
```

Comment:
i suggest writing it like this , since we are not using `getData` function anywhere else 

`const consumerGroups = React.useMemo(
    () => data.filter((item) => item.groupId.toLocaleLowerCase().indexOf(keyword) > -1),
    [data, keyword]
  );`